### ISSUE TITLE
adding source prevents geometry added to draw from rendering

### ISSUE THREAD

**Issue by demiurg** [mapbox/mapbox-gl-draw#724]
https://codepen.io/demiurge/pen/baNKLP?editors=1011

**mapbox-gl-js version**: 0.42.2
**mapbox-gl-draw version**: 1.0.4

### Steps to Trigger Behavior

 1. add a source
 2. add geometry to draw
 3. observe no geometry being rendered
 4. click draw square, observe geometry rendered

### Expected Behavior

should render

### Actual Behavior

does not render. However, if no source is added prior, OR moved below the draw code, things work as expected. This looks like a bug in order of something.

**Comment by herrethan** [comment]
I'm running into this very issue too!
In case this helps enlighten the situation, my hacky workaround for the time being is:
```
var onSourceReady = function(e){
  if(e.sourceId == 'mapbox-gl-draw-cold' && e.isSourceLoaded){
    map.off('sourcedata', onSourceReady);
    // draw.add() will properly render polygons now
  }
}
map.on('sourcedata', onSourceReady);
```

**Comment by herrethan** [comment]
looks like a race condition with sources or layer drawing of some kind, as I found wrapping the `draw.add()` in a loop checking for `map.loaded()` works

**Comment by mcwhittemore** [comment]
@demiurg - can you confirm if this is fixed by 1.0.7?

**Comment by demiurg** [comment]
Confirmed! Thanks a lot :)
